.github/workflows/publish-python-client.yml

```diff
@@ -162,12 +174,14 @@ jobs:
       - name: Remove osparc_client wheel
         run: rm -f dist/${{needs.build-n-publish.outputs.osparc_client}}
       - name: Publish osparc to Test PyPI
+        if: github.ref == 'refs/heads/master' && github.event_name == 'push'
         uses: pypa/gh-action-pypi-publish@release/v1
         with:
           repository-url: https://test.pypi.org/legacy/
           verbose: true
           packages-dir: dist/
       - name: Publish osparc to PyPI
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```

Comment:
Actually, is this protected anywhere that it will only run on master?

Reply:
I was looking into how this can be checked but I couldn't see how to easily do it. Tbh I didn't spend a lot of time checking it, but the point is that the `github.ref` variable which is available here does not tell if this tagged commit comes from master. Perhaps I could check above if the current `HEAD` is the latest commit on `master`. Or do you have any better ideas? What I did here is the suggested way in the docu for the `gh-action-pypi-publishing` packaging we are using here.

Reply:
Do you have a link to that docu? 
Maybe they only do it in an workflow that anyway only runs on master? (this file isn't as far as I can see?)

Reply:
Yes, sorry it is not actually that documentation, but a link which is in that documentation: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#defining-a-workflow-job-environment
I will investigate if there is an easy way to check this. But I think another point is that we should always be using forks for developing, so basically there should only be a single branch (=`master`) on the main repo. Not sure if that is currently the case though...

Reply:
I think on option is to protect the entire (or at least the relevant part of the file) file from only running on master:
https://github.com/BlueBrain/BluePyOpt/blob/5232a8f2e278cf0d32fe7008a81d82b0ad10b992/.github/workflows/build.yml#L3

Reply:
In my opinion, but I could be wrong, is to protect the entire file with:
`on:
  push:
    branches: [master]
`
and then protect the specific task with the startsWith(github.ref, 'refs/tags')

Reply:
Actually, also suggested here: https://stackoverflow.com/a/66463380

Reply:
Thanks. The thing is that, in the current setup, this workflow should also be run on PRs, so not only the master branch. Maybe I need to split the workflow into two workflows. I can try out what you suggest, but I am a bit skeptical about what they suggest in the stackoverflow link. As far as I can tell from https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches-and-tags the `on: push: branches: [master]` just evaluates against the git ref (available via) `github.ref` in the workflow. That will either be of the form `refs/head/<branch name>` or `refs/tags/<tag>`. So I (if I understand correctly the gh docu) `on: push: branches: [master]` is the same as `on: push` and then in the job testing for `github.ref == 'refs/heads/master`.

Reply:
Yeah, I think it definitely requires splitting the files in relevant parts. I think it anyway makes sense to have a separate workflow file that only runs on master.

Reply:
Makes sense, thanks for the feedback. I will try splitting out the publishing of the package into a separate workflow tomorrow morning

Reply:
@wvangeit I setup a new workflow for handling the publishing to Pypi.

Reply:
Very good @bisgaard-itis